reflect.go

```diff
@@ -415,19 +415,18 @@ func (ri *reflectInspector) recursivelyRecordUsedForReflect(t types.Type) {
 			}
 
 			// Record the field itself, too.
-			ri.recordUsedForReflect(field)
+			ri.recordUsedForReflect(field, t)
 
-			ri.recursivelyRecordUsedForReflect(field.Type())
+			ri.recursivelyRecordUsedForReflect(field.Type(), nil)
 		}
 
 	case interface{ Elem() types.Type }:
 		// Get past pointers, slices, etc.
-		ri.recursivelyRecordUsedForReflect(t.Elem())
+		ri.recursivelyRecordUsedForReflect(t.Elem(), nil)
 	}
 }
 
-// TODO: consider caching recordedObjectString via a map,
-// if that shows an improvement in our benchmark
+// TODO: figure out how to move `EmbeddedAliasFields` to reflectedObjectString
```

Comment:
EmbeddedAliasFields will go away very soon, in two months at the latest once 1.24.0 is out and we can use proper alias tracking, so IMO this TODO is not needed